src/core/Akka.Cluster/ClusterDaemon.cs

```diff
@@ -2465,6 +2493,35 @@ public bool ValidNodeForGossip(UniqueAddress node)
                     _latestGossip.ReachabilityExcludingDownedObservers.Value.IsReachable(node);
         }
 
+        /// <summary>
+        /// The Exiting change is gossiped to the two oldest nodes for quick dissemination to potential Singleton nodes
+        /// </summary>
+        /// <param name="latestGossip"></param>
+        /// <param name="exitingMembers"></param>
+        /// <returns></returns>
+        public static IEnumerable<Member> GossipTargetsForExitingMembers(Gossip latestGossip, IEnumerable<Member> exitingMembers)
+        {
+            if (exitingMembers.Any())
+            {
+                var roles = exitingMembers.SelectMany(m => m.Roles);
```

Comment:
Might be enumerating the `existingMembers` collection multiple times, but this doesn't appear to be an issue per the actual unit tests and MNTR results.